components/src/core/DropdownToolbar/DropdownToolbar.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useCallback, useRef } from 'react';
+import {
+    Typography,
+    Toolbar,
+    ListItemText,
+    Menu,
+    MenuItem,
+    ToolbarProps,
+    createStyles,
+    makeStyles,
+    useTheme,
+    Theme,
+} from '@material-ui/core';
+import { ArrowDropDown } from '@material-ui/icons';
+import clsx from 'clsx';
+
+const MENU_ITEM_HEIGHT = 48;
+
+export type ToolbarMenuItem = {
+    label: string;
+    onClick: () => void;
+};
+export type DropdownToolbarProps = ToolbarProps & {
+    title: string;
+    subtitle?: string;
+    menuItems?: ToolbarMenuItem[];
+    navigationIcon?: JSX.Element;
+    customMenu?: JSX.Element;
+    classes?: DropdownToolbarClasses;
```

Comment:
can we match the menu API from the UserMenu (eventually I want them to share a common menu component between them):
```suggestion
export type DropdownToolbarProps = ToolbarProps & {
    title: string;
    subtitle?: string;
    menuGroups?: ToolbarMenuGroup[];
    navigationIcon?: JSX.Element;
    menu?: JSX.Element;
    MenuProps?: MenuProps;
    onOpen?: Function;
    onClose?: Function;
    classes?: DropdownToolbarClasses;
```

Reply:
https://github.com/pxblue/react-component-library/issues/181